bootstrap/lib/cache.js

```diff
@@ -37,31 +50,58 @@ const withCache = (execute, options) => {
   // Algorithm we use to derive entry key
   const hashOptions = {
     algorithm: 'sha1',
+
     encoding: 'hex',
     excludeKeys: (key) => options.ignoredKeys.includes(key),
   }
+  const _getKey = (data) => hash(data, hashOptions)
+  const _getCoalescingKey = (key) => hash(`${options.requestCoalescingGroup}--${key}`, hashOptions)
 
   const _getMaxAge = (data) => {
     if (!data || !data.data) return options.maxAge
     if (isNaN(data.data.maxAge)) return options.maxAge
     return data.data.maxAge || options.maxAge
   }
 
-  return (data, callback) => {
-    const key = hash(data, hashOptions)
+  const _getWithCoalescing = async (key, coalescingKey, retries = 5, ms = 100) => {
+    if (retries === 0) return null
+    const entry = cache.get(key)
+    if (entry) {
+      logger.debug(`Request coalescing: Success`)
+      return entry
+    }
+    const inFlight = cache.get(coalescingKey)
+    if (!inFlight) return null
+    // Capped linear back-off: 100, 200, 400, 800, 1000..
+    ms = Math.min(ms * 2, 1000)
```

Comment:
Any reason for a retry? Imo retrying could result in delayed answers when we can just fallback to making a real request to the API instead.

Reply:
As per our offline discussion, request coalescing is a separate enhancement from the remote caching support.